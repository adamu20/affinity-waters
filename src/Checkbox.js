import React from 'react';

const Checkbox = ({ label, value, onChange }) => (
  <div>
    <label>
      <input type='checkbox' checked={value} onChange={onChange} />
      {label}
    </label>
  </div>
);

export default Checkbox;
