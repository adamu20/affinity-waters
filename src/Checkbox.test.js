import React from 'react';
import { shallow } from 'enzyme';
import Checkbox from './Checkbox';

describe('Checkbox', () => {
  let wrapped = shallow(<Checkbox />);
  it('should render the Checkbox Component correctly', () => {
    expect(wrapped).toMatchSnapshot();
  });

  it('simulate checkbox', () => {
    expect(wrapped.find('input')).toHaveLength(1);
    wrapped.find('input').simulate('change', { target: { checked: true } });
    wrapped.update();
    expect(wrapped.find('input').props().checked).toBe(true);
  });
});
