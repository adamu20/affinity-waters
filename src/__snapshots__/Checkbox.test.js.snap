// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox should render the Checkbox Component correctly 1`] = `
<div>
  <label>
    <input
      type="checkbox"
    />
  </label>
</div>
`;
